src/packages/external-source/external-source.ts

```diff
@@ -0,0 +1,497 @@
+
+  const headers: HeadersInit = {
+    Authorization: authorizationHeader ?? '',
+    'Content-Type': 'application/json',
+    'x-hasura-role': roleHeader ? `${roleHeader}` : '',
+    'x-hasura-user-id': userHeader ? `${userHeader}` : '',
+  };
+
+  // Validate uploaded attribute schemas are formatted validly
+  const schemasAreValid: boolean = await compiledAttributeMetaschema({
+    event_types: parsedEventTypes,
+    source_types: parsedSourceTypes,
+  });
+  if (!schemasAreValid) {
+    logger.error(`POST /uploadExternalSourceEventTypes: Schema validation failed for uploaded source and event types.`);
+    compiledAttributeMetaschema.errors?.forEach(error => logger.error(error));
+    res.status(500).send({ message: compiledAttributeMetaschema.errors });
+    return;
+  }
+
+  logger.info(`POST /uploadExternalSourceEventTypes: Uploaded attribute schema(s) are VALID`);
+
+  // extract the external sources and event types
+  const externalSourceTypeInput: ExternalSourceTypeInsertInput[] = [];
+  const externalEventTypeInput: ExternalEventTypeInsertInput[] = [];
+
+  const eventTypeKeys = Object.keys(parsedEventTypes);
+  for (const externalEventType of eventTypeKeys) {
+    externalEventTypeInput.push({
+      attribute_schema: parsedEventTypes[externalEventType],
+      name: externalEventType,
+    });
+  }
+
+  const sourceTypeKeys = Object.keys(parsedSourceTypes);
+  for (const externalSourceType of sourceTypeKeys) {
+    externalSourceTypeInput.push({
+      attribute_schema: parsedSourceTypes[externalSourceType],
+      name: externalSourceType,
+    });
+  }
+
+  // Run the Hasura migration for creating all types, in one go
+  const response = await fetch(GQL_API_URL, {
+    body: JSON.stringify({
+      query: gql.CREATE_EXTERNAL_SOURCE_EVENT_TYPES,
+      variables: { externalEventTypes: externalEventTypeInput, externalSourceTypes: externalSourceTypeInput },
+    }),
+    headers,
+    method: 'POST',
+  });
+
+  const createExternalSourceEventTypesResponse = await response.json();
+  if (createExternalSourceEventTypesResponse?.data !== undefined) {
+    res.json(createExternalSourceEventTypesResponse.data as CreateExternalSourceEventTypeResponse);
+  } else {
+    res.json(createExternalSourceEventTypesResponse as HasuraError);
+  }
+}
+
+async function uploadExternalSource(req: Request, res: Response) {
+  const authorizationHeader = req.get('authorization');
+
+  const {
+    headers: { 'x-hasura-role': roleHeader, 'x-hasura-user-id': userHeader },
+  } = req;
+
+  const { body } = req;
+
+  if (typeof body !== 'object') {
+    logger.error(
+      `POST /uploadExternalSourceEventTypes: Body of request must be a JSON, with two stringified properties: "source" and "events".`,
+    );
+    res
+      .status(500)
+      .send({ message: `Body of request must be a JSON, with two stringified properties: "source" and "events".` });
+    return;
+  }
```

Comment:
This `if` statement is unnecessary. Additionally, this error message describes a **different body**  than the swagger docs. Please align the two. (Additionally please move the error string into a const if you're going to keep repeating it)

Reply:
Removed the `if` and fixed the swagger docs for both end-points. Also went through and all `logger` + `res.status(500).send({...})` combos are now using a consistent `errorMsg` for both!